salt/grains/core.py

```diff
@@ -1168,6 +1168,9 @@ def _clean_value(key, val):
         # AssetTag0. Manufacturer04. Begone.
         if re.search(r'manufacturer|to be filled|available|asset|^no(ne|t)', val, flags=re.IGNORECASE):
             return None
+    elif key == 'chassis_type':
+        if val in [0x02, '0x02', '2', 'Unknown']:
+            return None
```

Comment:
You don't need to explicitly write `2` in hexadecimal, as Python will implicitly convert it during the comparison, so just `2` is fine (the `2 == 0x02` or `2 in [0x02]` and vice versa returns `True`). However, this sort of list isn't reliable and prone to bugs, especially if `chassis_type` appears to be like `0xFF` or `0` or whatever proprietary hardware will return you — it also should be considered as unknown (it is above `0x1D`), etc. Hence I would write it this way:

```python
try:
    val = int(val, 16)
    if not val or val == 2 or val > max(chassis_types):
        val = None
except ValueError:
        val = None
```

Since now `val` is already either value or `None`, no need to explicitly `return None` as it will anyway fall off to the last `return`. I know you've just copied the previous code, but [multiple returns aren't nice](http://docs.python-guide.org/en/latest/writing/style/#returning-values) (and so the older code too).

Reply:
@isbm Please have another look at the list of Values I consider unknown. I did not check for 2 or 0x02 which would indeed be duplicate. I specifically checked for 0x02 and 2 as string values in case someone somewhere forgot a cast. The case of int(2) is already covered with the first val 0x02.

And if you look at https://github.com/saltstack/salt/pull/48273/commits/b78ba338a42aa28f54f6dfb9a476f97610da4c22#diff-6e894b39b67c4bab67796752744ea9d5R2418 the case of unknown value like 0xff or -1 or 0 is already covered as well.

Reply:
@ixs I would prefer you just cast everything, regardless who forgot what, instead of creating a special hard-coded list of things that you assume you've got right.